lib/trino-filesystem/src/main/java/io/trino/filesystem/Location.java

```diff
@@ -96,7 +96,10 @@ public static Location of(String location)
                 }
             }
 
-            String path = (authoritySplit.size() == 2) ? authoritySplit.get(1) : "";
+            if (authoritySplit.size() != 2) {
+                throw new IllegalArgumentException("Path missing in file system location: " + location);
```

Comment:
```
trino> show create table hive.default.findinpathtop1;
                      Create Table                       
---------------------------------------------------------
 CREATE TABLE hive.default.findinpathtop1 (              
    c varchar                                            
 )                                                       
 WITH (                                                  
    external_location = 's3://trino-findinpath', 
    format = 'TEXTFILE'                                  
 )                                                       
(1 row)

Query 20230616_084756_00004_wjnqc, FINISHED, 1 node
http://localhost:8080/ui/query.html?20230616_084756_00004_wjnqc
Splits: 1 total, 1 done (100.00%)
CPU Time: 0.0s total,     0 rows/s,     0B/s, 100% active
Per Node: 0.0 parallelism,     0 rows/s,     0B/s
Parallelism: 0.0
Peak Memory: 450B
1.60 [0 rows, 0B] [0 rows/s, 0B/s]

trino> 
trino> select * from hive.default.findinpathtop1;
Query 20230616_084811_00005_wjnqc failed: Path missing in file system location: s3://trino-findinpath
io.trino.spi.TrinoException: Path missing in file system location: s3://trino-findinpath
	at io.trino.plugin.hive.BackgroundHiveSplitLoader$HiveSplitLoaderTask.process(BackgroundHiveSplitLoader.java:318)
	at io.trino.plugin.hive.util.ResumableTasks$1.run(ResumableTasks.java:38)
	at io.trino.$gen.Trino_dev____20230616_084703_2.run(Unknown Source)
	at io.airlift.concurrent.BoundedExecutor.drainQueue(BoundedExecutor.java:80)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1589)
Caused by: java.lang.IllegalArgumentException: Path missing in file system location: s3://trino-findinpath
	at io.trino.filesystem.Location.of(Location.java:100)
	at io.trino.plugin.hive.BackgroundHiveSplitLoader.loadPartition(BackgroundHiveSplitLoader.java:551)
```


We figure out rather late that we're dealing with an invalid table location.
Shouldn't we do this check much earlier?

Reply:
How did you create this table?
In the test it's seen that CREATE TABLE rejects such location.

Reply:
I created it yesterday evening while testing the current `master`.
The easiest way to repro is to strip from Glue `/` from `s3://bucket/`
Feel free to ignore my comment.

Reply:
> I created it yesterday evening while testing the current `master`.

Yes, master allows creation of broken tables (as shown in the commit adding the test)